<?xml version="1.0"?>

<robot name="robot_arm_new">
  <link name="world"/>
  <link name="arm_1">
    <inertial>
      <origin xyz="7.7128E-09 -0.063005 -3.01969999961422E-08" rpy="0 0 0" />
      <mass
        value="1.6004" />
      <inertia ixx="6.88524389" ixy="7.9550614501301E-10" ixz="-1.34378458924839E-09" iyy="4.39398313"
        iyz="-1.10071809773382E-08"
        izz="0.00553739792746489" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 1.57079633" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/arm_base.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 1.57079633" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/arm_base.stl" />
      </geometry>
    </collision>
</link>


<link name="arm_2">
    <inertial>
      <origin xyz="0.12821 3.5589E-06 0.052492" rpy="0 0 0" />
      <mass value="0.200" />
      <inertia ixx="0.000222789" ixy="-9.6452938E-8" ixz="-0.00023482764" iyy="0.001302429" iyz="-3.31952134E-8" izz="0.00121768172" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0.05166"
        rpy="1.57079633 0 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/bicep.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0.05166"
        rpy="1.57079633 0 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/bicep.stl" />
      </geometry>
    </collision>
  </link>


  <link
    name="arm_3">
    <inertial>
      <origin
        xyz="-9.1053E-08 -0.069257 -1.86629999995759E-07"
        rpy="0 0 0" />
      <mass
        value="0.27721" />
      <inertia
        ixx="0.00104290750143942"
        ixy="4.37155302268076E-09"
        ixz="-2.45049603914627E-09"
        iyy="0.000380518373895034"
        iyz="-7.56009835172156E-09"
        izz="0.00106006525067445" />
    </inertial>
    <visual>
      <origin
        xyz="0.1359 0 0"
        rpy="0 -1.570796326 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/wrist.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0.1359 0 0"
        rpy="0 -1.570796326 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/wrist.stl" />
      </geometry>
    </collision>
  </link>


 <link
    name="arm_adapter1">
    <inertial>
      <origin
        xyz="-0.11109 1.1476E-08 0.046469"
        rpy="0 0 0" />
      <mass
        value="0.84845" />
      <inertia
        ixx="0.00079656"
        ixy="-7.8011E-10"
        ixz="0.00053616"
        iyy="0.003576"
        iyz="4.6326E-10"
        izz="0.0033698" />
    </inertial>
    <visual>
      <origin
        xyz="0.20994 0 0.05163"
        rpy="1.570796326 0 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/elbow.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0.20994 0 0.05163"
        rpy="1.570796326 0 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/elbow.stl" />
      </geometry>
    </collision>
  </link>

<link
    name="arm_4">
    <inertial>
      <origin
        xyz="-9.1053E-08 -0.069257 -1.86629999995759E-07"
        rpy="0 0 0" />
      <mass
        value="0.27721" />
      <inertia
        ixx="0.00104290750143942"
        ixy="4.37155302268076E-09"
        ixz="-2.45049603914627E-09"
        iyy="0.000380518373895034"
        iyz="-7.56009835172156E-09"
        izz="0.00106006525067445" />
    </inertial>
    <visual>
      <origin
        xyz="0.13522 0 0"
        rpy="3.14159 1.570796326 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/wrist.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0.13522 0 0"
        rpy="3.14159 1.570796326 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/wrist.stl" />
      </geometry>
    </collision>
  </link>
<link
    name="arm_adapter2">
    <inertial>
      <origin
        xyz="-0.11109 1.1476E-08 0.046469"
        rpy="0 0 0" />
      <mass
        value="0.84845" />
      <inertia
        ixx="0.00079656"
        ixy="-7.8011E-10"
        ixz="0.00053616"
        iyy="0.003576"
        iyz="4.6326E-10"
        izz="0.0033698" />
    </inertial>
    <visual>
      <origin
        xyz="0.20994 0.05163 0"
        rpy="0 0 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/elbow.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0.20994 0.05163 0"
        rpy="0 0 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/elbow.stl" />
      </geometry>
    </collision>
  </link>
<link
    name="arm_5">
    <inertial>
      <origin
        xyz="-9.1053E-08 -0.069257 -1.86629999995759E-07"
        rpy="0 0 0" />
      <mass
        value="0.27721" />
      <inertia
        ixx="0.00104290750143942"
        ixy="4.37155302268076E-09"
        ixz="-2.45049603914627E-09"
        iyy="0.000380518373895034"
        iyz="-7.56009835172156E-09"
        izz="0.00106006525067445" />
    </inertial>
    <visual>
      <origin
        xyz="0.13522 0 0"
        rpy="0 -1.570796326 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/wrist.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0.13522 0 0"
        rpy="0 -1.570796326 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/wrist.stl" />
      </geometry>
    </collision>
  </link>
<link
    name="arm_adapter3">
    <inertial>
      <origin
        xyz="-9.1053E-08 -0.069257 -1.86629999995759E-07"
        rpy="0 0 0" />
      <mass
        value="0.27721" />
      <inertia
        ixx="0.00104290750143942"
        ixy="4.37155302268076E-09"
        ixz="-2.45049603914627E-09"
        iyy="0.000380518373895034"
        iyz="-7.56009835172156E-09"
        izz="0.00106006525067445" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 1.570796326 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/wrist.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 1.570796326 0" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/wrist.stl" />
      </geometry>
    </collision>
  </link>
<link
    name="arm_6">
    <inertial>
      <origin
        xyz="-0.11109 1.1476E-08 0.046469"
        rpy="0 0 0" />
      <mass
        value="0.84845" />
      <inertia
        ixx="0.00079656"
        ixy="-7.8011E-10"
        ixz="0.00053616"
        iyy="0.003576"
        iyz="4.6326E-10"
        izz="0.0033698" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0.05163"
        rpy="1.57079633 0 3.14159" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/elbow.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0.05163"
        rpy="1.57079633 0 3.14159" />
      <geometry>
        <mesh filename="package://robotic_arm/meshes/elbow.stl" />
      </geometry>
    </collision>
  </link>
<!--######################################################################-->
<gazebo reference="arm_1">
  <selfCollide>false</selfCollide>
</gazebo>

<gazebo reference="arm_2">
  <selfCollide>false</selfCollide>
</gazebo>

<gazebo reference="arm_3">
  <selfCollide>false</selfCollide>
</gazebo>

<gazebo reference="arm_4">
  <selfCollide>false</selfCollide>
</gazebo>


<gazebo reference="arm_5">
  <selfCollide>false</selfCollide>
</gazebo>

<gazebo reference="arm_6">
  <selfCollide>false</selfCollide>
</gazebo>


<gazebo reference="arm_adapter_1">
  <selfCollide>false</selfCollide>
</gazebo>


<gazebo reference="arm_adapter_2">
  <selfCollide>false</selfCollide>
</gazebo>

<gazebo reference="arm_adapter_3">
  <selfCollide>false</selfCollide>
</gazebo>
<!--######################################################################-->
<joint name="arm_1_joint" type="continuous">
  <axis xyz="0 0 1"/>
  <parent link="world"/>
  <child link="arm_1"/>
  <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
</joint>


<joint name="arm_2_joint" type="continuous">
  <axis xyz="0 0 1"/>
  <parent link="arm_1"/>
  <child link="arm_2"/>
  <origin rpy="1.570796326 0 0" xyz="0 0.0 0.20271"/>
</joint>

<joint name="arm_3_joint" type="continuous">
  <axis xyz="0 0 1"/>
  <parent link="arm_2"/>
  <child link="arm_3"/>
  <origin rpy="0 0 0" xyz="0.269 0.0 0"/>
</joint>

<joint name="arm_adapter1_axis" type="fixed">
  <axis xyz="0 0 1"/>
  <parent link="arm_3"/>
  <child link="arm_adapter1"/>
  <origin rpy="0 0 0" xyz="0.13522 0 0"/>
</joint>


<joint name="arm_4_joint" type="continuous">
  <axis xyz="0 0 1"/>
  <parent link="arm_3"/>
  <child link="arm_4"/>
  <origin rpy="0 0 0" xyz="0.34516 0 0"/>
</joint>

<joint name="arm_adapter2_axis" type="fixed">
  <axis xyz="0 0 1"/>
  <parent link="arm_4"/>
  <child link="arm_adapter2"/>
  <origin rpy="0 0 0" xyz="0.13522 0 0"/>
</joint>

<joint name="arm_5_joint" type="continuous">
  <axis xyz="0 0 1"/>
  <parent link="arm_4"/>
  <child link="arm_5"/>
  <origin rpy="-1.570796326 0 0" xyz="0.34516 0 0"/>
</joint>

<joint name="arm_adapter3_axis" type="fixed">
  <axis xyz="0 0 1"/>
  <parent link="arm_5"/>
  <child link="arm_adapter3"/>
  <origin rpy="1.570796326 0 0" xyz="0.13522 0 0"/>
</joint>

<joint name="arm_6_joint" type="continuous">
  <axis xyz="0 0 1"/>
  <parent link="arm_5"/>
  <child link="arm_6"/>
  <origin rpy="1.570796326 0 0" xyz="0.27 0 0"/>
</joint>


<!--######################################################################-->
<transmission name="arm_1_trans" type="SimpleTransmission">
  <type>transmission_interface/SimpleTransmission</type>
  <actuator name="arm_1_motor">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
   <mechanicalReduction>1</mechanicalReduction>
  </actuator>
  <joint name="arm_1_joint">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
  </joint>
</transmission>


<transmission name="arm_2_trans" type="SimpleTransmission">
  <type>transmission_interface/SimpleTransmission</type>
  <actuator name="arm_2_motor">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
   <mechanicalReduction>1</mechanicalReduction>
  </actuator>
  <joint name="arm_2_joint">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
  </joint>
</transmission>


<transmission name="arm_3_trans" type="SimpleTransmission">
  <type>transmission_interface/SimpleTransmission</type>
  <actuator name="arm_3_motor">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
   <mechanicalReduction>1</mechanicalReduction>
  </actuator>
  <joint name="arm_3_joint">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
  </joint>
</transmission>

<transmission name="arm_4_trans" type="SimpleTransmission">
  <type>transmission_interface/SimpleTransmission</type>
  <actuator name="arm_4_motor">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
   <mechanicalReduction>1</mechanicalReduction>
  </actuator>
  <joint name="arm_4_joint">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
  </joint>
</transmission>


<transmission name="arm_5_trans" type="SimpleTransmission">
  <type>transmission_interface/SimpleTransmission</type>
  <actuator name="arm_5_motor">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
   <mechanicalReduction>1</mechanicalReduction>
  </actuator>
  <joint name="arm_5_joint">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
  </joint>
</transmission>


<transmission name="arm_6_trans" type="SimpleTransmission">
  <type>transmission_interface/SimpleTransmission</type>
  <actuator name="arm_6_motor">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
   <mechanicalReduction>1</mechanicalReduction>
  </actuator>
  <joint name="arm_6_joint">
   <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
  </joint>
</transmission>

<!--######################################################################-->
<gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotNamespace>/</robotNamespace>
      <controlPeriod>0.001</controlPeriod>
      <legacyModeNS>false</legacyModeNS>
    </plugin>
  </gazebo>
<!--######################################################################-->
  <gazebo>
    <plugin name="joint_state_publisher" filename="libgazebo_ros_joint_state_publisher.so">
      <jointName>arm_1_joint, arm_2_joint, arm_3_joint, arm_4_joint, arm_5_joint, arm_6_joint</jointName>
    </plugin>
  </gazebo>
<!--######################################################################-->


</robot>